pkg/alerter/klaxon.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alerter
+
+import (
+	"context"
+
+	configpb "github.com/GoogleCloudPlatform/testgrid/pb/config"
+	summarypb "github.com/GoogleCloudPlatform/testgrid/pb/summary"
+)
+
+// A Klaxon is a component that reacts when a difference in summary is detected
+type Klaxon interface {
+	// OnConsistentFailure is called when a DashboardTab has failed <num_alerts> times where it hasn't before.
+	// OnConsistentFailure won't be called again until the test succeeds once.
+	OnConsistentFailure(ctx context.Context, new, old *summarypb.DashboardTabSummary, config *configpb.DashboardTab) error
+
+	// OnEachFailure is called for every loop that the DashboardTab is summarized as failed.
+	// OnEachFailure is called over and over again, so it's important that it be idempotent.
+	OnEachFailure(ctx context.Context, new, old *summarypb.DashboardTabSummary, config *configpb.DashboardTab) error
+
+	// OnEachSuccess is called for every loop that the DashboardTab is summarized as passing
+	// OnEachSuccess is called over and over again.
+	OnEachSuccess(ctx context.Context, new, old *summarypb.DashboardTabSummary, config *configpb.DashboardTab) error
```

Comment:
Why do we need/want to call this? Why wouldn't we use something like pubsub for it?

Reply:
A disadvantage of using an interface like this is that adding a function breaks older instances of the interface. Moving this to pubsub might be the right move later, but for now I think that would add complexity for no benefit.

The idea behind this function is to cover a use case like "When the test stops failing, alert me that everything's fine now".